bleak/backends/client.py

```diff
@@ -118,8 +118,8 @@ async def unpair(self) -> bool:
         """Unpair with the peripheral."""
         raise NotImplementedError()
 
-    @abc.abstractmethod
-    async def is_connected(self) -> bool:
+    @abc.abstractproperty
+    def is_connected(self) -> bool:
```

Comment:
This is a potential breaking change in the API. I do agree to the change, since we do not query anything from the OS in any backend regarding the connection status but rely entirely on callbacks. We need to remember to document this and tell potential issue creators. Could we have a async method with a `PendingDeprecationWarning` or `DeprecationWarning` warning calling the sync method?

Reply:
I added a commit for this. I went with `FutureWarning` to be sure all users will see it since the ones you suggested are ignored by default.